src/main/java/ch/njol/skript/events/EvtCommand.java

```diff
@@ -18,66 +18,136 @@
  */
 package ch.njol.skript.events;
 
+import org.bukkit.command.CommandSender;
+import org.bukkit.command.ConsoleCommandSender;
+import org.bukkit.entity.Player;
 import org.bukkit.event.Event;
 import org.bukkit.event.player.PlayerCommandPreprocessEvent;
 import org.bukkit.event.server.ServerCommandEvent;
 import org.eclipse.jdt.annotation.Nullable;
 
 import ch.njol.skript.Skript;
+import ch.njol.skript.SkriptConfig;
 import ch.njol.skript.lang.Literal;
 import ch.njol.skript.lang.SkriptEvent;
 import ch.njol.skript.lang.SkriptParser.ParseResult;
 import ch.njol.util.StringUtils;
 import ch.njol.util.coll.CollectionUtils;
 
-/**
- * @author Peter Güttinger
- */
-@SuppressWarnings("unchecked")
-public class EvtCommand extends SkriptEvent { // TODO condition to check whether a given command exists, & a conditon to check whether it's a custom skript command
+public class EvtCommand extends SkriptEvent {
+
 	static {
-		Skript.registerEvent("Command", EvtCommand.class, CollectionUtils.array(PlayerCommandPreprocessEvent.class, ServerCommandEvent.class), "command [%-string%]")
-				.description("Called when a player enters a command (not necessarily a Skript command) but you can check if command is a skript command, see <a href='conditions.html#CondIsSkriptCommand'>Is a Skript command condition</a>.")
-				.examples("on command:", "on command \"/stop\":", "on command \"pm Njol \":")
-				.since("2.0");
+		Skript.registerEvent("Command", EvtCommand.class, CollectionUtils.array(PlayerCommandPreprocessEvent.class, ServerCommandEvent.class),
+				"[1:console|2:player] command [%-string%]")
+				.description(
+						"Called when a player or the console enters a command (not necessarily a Skript command)" +
+						"but you can check if command is a skript command, see <a href='conditions.html#CondIsValidCommand'>Is Valid Command condition</a>.")
+				.examples(
+						"on command:",
+						"on command \"/stop\":",
+						"on command \"pm Njol\":",
+						"on console command:",
+						"on player command \"/op\"")
+				.keywords("console", "player")
+				.since("2.0, INSERT VERSION (specific executor)");
 	}
-	
+
+	private enum Sender {
+
+		ANY(""),
+		CONSOLE("console"),
+		PLAYER("player");
+
```

Comment:
Isn't there another pull request with an enum like this already?

Reply:
Well I'm not really sure it's doing much as an enum at all, if there already was one I would say use that but I'm not sure it's necessary.

It should also be lower down in the file.

Reply:
I have an alternative implementation for such if you prefer to not use the same enum implementation here and will let you know at a later time today.